mage/vars.go

```diff
@@ -287,40 +292,24 @@ func buildFlags() []string {
 	}
 }
 
-func importEnvFromMakefile() (env environ) {
-	env = make(environ)
+func importEnvFromMakefile() (env map[string]string) {
 	cmd := exec.Command("make", "-f", "Makefile.buildx", "magnet-vars")
 	out, err := cmd.CombinedOutput()
 	if err != nil {
-		log.WithError(err).Warn("Failed to import environ from makefile.")
-		return nil
+		panic(fmt.Sprint("failed to import environ from makefile:", err))
 	}
-	for _, line := range strings.Split(string(out), "\n") {
-		cols := strings.SplitN(line, "=", 2)
-		if len(cols) != 2 || !strings.HasPrefix(cols[0], "MAGNET_") {
-			log.Debug("Skip line that does not look like magnet envar.")
-			continue
-		}
-		key, value := strings.TrimPrefix(cols[0], "MAGNET_"), cols[1]
-		env[key] = value
+	env, err = magnet.ImportEnvFromReader(bytes.NewReader(out))
+	if err != nil {
+		panic(fmt.Sprint("failed to import environ from makefile:", err))
 	}
 	return env
 }
 
-func (r environ) getBuildDir() string {
-	return r["BUILDDIR"]
+func isPlainProgress() *bool {
```

Comment:
The idea is that magnet could apply automatic configuration (configuration by convention) via a set of implicit rules - e.g. it might look at the `DEBIAN_FRONTEND` environment variable to decide which log output mode to use. If this were a plain bool I'd have no way of telling whether I wanted it explicitly off or it was uninitialized.

Now that I'm thinking of it - it did grow out of a simple 2-state context to be 3-state so it might warrant a change from a boolean to the mode with multiple values but I'll leave that for later.